crates/wasi-runtime-config/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::Result;
+use std::collections::HashMap;
+
+mod gen_ {
+    wasmtime::component::bindgen!({
+        path: "wit",
+        world: "wasi:config/imports",
+        trappable_imports: true,
+    });
+}
+use self::gen_::wasi::config::runtime as generated;
+
+/// Capture the state necessary for use in the `wasi-runtime-config` API implementation.
+pub struct WasiRuntimeConfigCtx {
+    runtime_config: HashMap<String, String>,
+}
+
+impl WasiRuntimeConfigCtx {
+    /// Create a new context.
+    pub fn new<S, I>(config: I) -> Self
+    where
+        S: Into<String>,
+        I: IntoIterator<Item = (S, S)>,
+    {
+        Self {
+            runtime_config: config
+                .into_iter()
+                .map(|(k, v)| (k.into(), v.into()))
+                .collect(),
+        }
+    }
+}
+
+/// A wrapper capturing the needed internal `wasi-runtime-config` state.
+pub struct WasiRuntimeConfigView<'a> {
+    ctx: &'a WasiRuntimeConfigCtx,
+}
+
+impl<'a> WasiRuntimeConfigView<'a> {
+    /// Create a new view into the `wasi-runtime-config` state.
+    pub fn new(ctx: &'a WasiRuntimeConfigCtx) -> Self {
```

Comment:
And it would also be nice, for convenience, to

```rust
impl<'a> From<&'a WasiRuntimeConfigVariables> for WasiRuntimeConfig<'a> {
    // ...
}
```

You might also want to consider

```rust
impl WasiRuntimeConfig<'_> {
    /// Create a new, empty runtime configuration.
    pub fn empty() -> WasiRuntimeConfig<'static> {
        static EMPTY_VARS: WasiRuntimeConfigVariables { /* ... */ };
        Self::new(&EMPTY_VARS)
    }
}
```